site/components/Navigation/NavItemMobile.tsx

```diff
@@ -1,26 +1,49 @@
-import Link from "next/link";
 import React, { FC } from "react";
 import * as styles from "./styles";
-
+import { cx } from "@emotion/css";
+import { ButtonPrimary } from "@klimadao/lib/components";
+import { NavItemMobileID } from "./index";
 interface Props extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
   name: string;
-  url: string;
-  number?: string;
+  id?: NavItemMobileID;
+  url?: string;
+  toggledNavItemID?: string | undefined;
+  setToggledNavItemID?: (string: NavItemMobileID | undefined) => void;
+  subMenu?: JSX.Element[];
+  active?: boolean;
 }
 
 export const NavItemMobile: FC<Props> = (props) => {
+  const isToggled =
+    !!props.id &&
+    !!props.toggledNavItemID &&
+    props.id === props.toggledNavItemID;
+  const onToggleSubmenu = () => {
+    if (isToggled && props.setToggledNavItemID)
+      return props.setToggledNavItemID(undefined);
+    props.setToggledNavItemID &&
+      props.id &&
+      props.setToggledNavItemID(props.id);
+  };
+
   return (
-    <div className={styles.navMain_MobileItem}>
-      <Link href={props.url}>
-        <a {...props} className={styles.navMain_MobileLink}>
-          <div className={styles.navMain_MobileItemInner}>
-            <span className={styles.navMain_MobileItemInnerNumber}>
-              {props.number}
-            </span>
-            {props.name}
-          </div>
-        </a>
-      </Link>
+    <div className={cx(styles.navMain_MobileItem)}>
+      <ButtonPrimary
```

Comment:
we need to pass `Link` down so that it uses nextjs built-in prefetching and localization.
```suggestion
      <ButtonPrimary
        link={Link}
```